control/controlhttp/client.go

```diff
@@ -255,7 +295,7 @@ func (a *Dialer) dial(ctx context.Context) (*ClientConn, error) {
 
 	// If we get here, then we didn't get anywhere with our dial plan; fall back to just using DNS.
 	a.logf("controlhttp: failed dialing using DialPlan, falling back to DNS; errs=%s", merr.Error())
-	return a.dialHost(ctx, netip.Addr{})
+	return a.dialControlIPCandidate(ctx, tailcfg.ControlIPCandidate{})
 }
 
 // The TS_FORCE_NOISE_443 envknob forces the controlclient noise dialer to
```

Comment:
@bradfitz there is a potential pickle here too that I need to understand in order to fix.  As I understand it, I need to change `TS_FORCE_NOISE_443` and `forceNoise443` to `TS_FORCE_NOISE_NOT_80` and `forceNoiseNot80` respectively, and change the logic to only avoid port 80 (or perhaps ignore the bug and do what the SRV record says anyway in this case?).  This is assuming that the Docker Desktop VPNKit bug A) still exists 7 years later and B) is linked to port 80 implicitly and not plain http itself. It turns out you are the author of these changes, do you happen to recall the details?